src/core/public/legacy_platform/legacy_platform_service.test.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const mockLoadOrder: string[] = [];
+
+jest.mock('ui/metadata', () => {
+  mockLoadOrder.push('ui/metadata');
+  return {
+    __newPlatformInit__: jest.fn(),
+  };
+});
+
+jest.mock('ui/chrome', () => {
+  mockLoadOrder.push('ui/chrome');
+  return {
+    bootstrap: jest.fn(),
+  };
+});
+
+jest.mock('ui/test_harness', () => {
+  mockLoadOrder.push('ui/test_harness');
+  return {
+    bootstrap: jest.fn(),
+  };
+});
+
+import { LegacyPlatformService } from './legacy_platform_service';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  injectedMetadataStartContract.getLegacyMetadata.mockReset();
+  jest.resetModules();
+  mockLoadOrder.length = 0;
+});
+
+const injectedMetadataStartContract = {
+  getLegacyMetadata: jest.fn(),
+};
+
+const requireLegacyFiles = jest.fn(() => {
+  mockLoadOrder.push('legacy files');
+});
+
+describe('#start()', () => {
+  describe('default', () => {
+    it('does not return a start contract', () => {
```

Comment:
I'm not sure if it's worth testing this. Generally we wouldn't treat a function's return type going from nothing -> something as a breaking change. I don't have a strong opinion that we shouldn't, though.

Reply:
Yeah, I agree. I was kind of unsure how much to test and how much to rely on TypeScript. It feels like a lot of the stuff here is handled by TypeScript, but 🤷‍♂️